File: leetcode/11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 29/04/2026, 23:14:23
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1

        res = 0

        while l < r:
            if height[l] < height[r]:
                res = max(res,(r-l) * height[l])
                l += 1
            else:
                res = max(res,(r-l) * height[r])
                r -= 1
        return res

File: leetcode/11-ContainerWithMostWater/11-ContainerWithMostWater.java
// Last updated: 29/04/2026, 23:14:29
class Solution {
    public int maxArea(int[] height) {
        int start=0;
        int end=height.length-1;
        int maxWater = 0;
        while(start<end) {
            int waterHere = Math.min(height[start], height[end]) * (end-start);
            maxWater = Math.max(waterHere, maxWater);
            if(height[start]<height[end]) {
                start+=1;
            } else {
                end-=1;
            }
        }
        return maxWater;
    }
}